src/Parser/Priest/Discipline/Modules/Features/AtonementHealingDone.js

```diff
@@ -47,6 +60,15 @@ class AtonementHealingDone extends Analyzer {
     this.bySource[spellId] = this.bySource[spellId] || {};
     this.bySource[spellId].ability = ability;
     this.bySource[spellId].healing = (this.bySource[spellId].healing || new HealingValue()).add(amount, absorbed, overheal);
+
+    if(spellId === SPELLS.PENANCE.id) {
+      if(!this.bySource[SPELLS.PENANCE.id].bolts) {
+        this.bySource[SPELLS.PENANCE.id].bolts = [0,0,0,0];
```

Comment:
While I don't expect Penance to gain any bolts any time soon, I think it would be better to write this without having to hard-code the amount of bolts. How about `this.bySource[SPELLS.PENANCE.id].bolts[this._lastPenanceBoltNumber] = (this.bySource[SPELLS.PENANCE.id].bolts[this._lastPenanceBoltNumber] || 0) + amount + (absorbed || 0);`?
ps. amount is always defined, absorbed may be undefined. The old method would not include any healing with absorbed not defined.